File: quartz-energy/src/backend/QuartzEnergy.FracSchedule.Services/Vega/Models/Vehicles/Overviews/VehicleOverviews.cs
namespace QuartzEnergy.FracSchedule.Services.Vega.Models.Vehicles.Overviews
{
    using System.Collections.Generic;

    using QuartzEnergy.Common.Services.Models.Overviews;

    public sealed class VehicleOverviews : OverviewsModel<VehicleOverviewsRequest, VehicleOverview>
    {
        public VehicleOverviews(
            VehicleOverviewsRequest request, 
            int recordsCount, 
            IEnumerable<VehicleOverview> overviews)
            : base(request, recordsCount, overviews)
        {
        }
    }
}

File: quartz-energy/src/frontend/src/app/vega/models/vehicles/overviews/vehicle-overviews.ts
import { VehicleOverviewsRequest } from './vehicle-overviews-request';
import { VehicleOverview } from './vehicle-overview';
import { Overviews } from '../../../../common/models/overviews/overviews';

export class VehicleOverviews extends Overviews<VehicleOverviewsRequest, VehicleOverview> {
    constructor(
      public request: VehicleOverviewsRequest,
      public recordsCount: number,
      public overviews: VehicleOverview[]) {
      super(request, recordsCount, overviews);
    }
}
